docs/proposals/06-immutable-etcd-backups.md

```diff
@@ -0,0 +1,315 @@
+
+#### Handling of Hibernated Clusters
+
+When an ETCD cluster is hibernated for a duration exceeding the immutability period, backups may become mutable again (this behavior depends on the cloud provider; refer to [Comparison of Storage Provider Properties](#comparison-of-storage-provider-properties-for-bucket-level-and-object-level-immutability)), compromising the intended immutability guarantees.
+
+Such handling of hibernated clusters is the type of scenario which the etcd operator-tasks frameworks lends itself to quite well, and thus for all proposed solutions, the operator tasks framework as defined [here](./05-etcd-operator-tasks.md) will be made use of for the designs of the solutions.
+
+To maintain snapshot immutability during extended hibernation, we propose two approaches:
+
+##### Approach 1: Using the Compaction Job
+
+**Proposed Solution:**
+
+Utilize the compaction job to periodically take fresh snapshots during hibernation. Introduce a new flag `--hibernation-snapshot-interval` to the compaction controller. This flag sets the interval after which a compaction job should be triggered for a hibernated ETCD cluster, based on the time elapsed since `fullLease.Spec.RenewTime.Time` and if `etcd.spec.replicas` is `0` (indicating hibernation). The compaction job uses the [compact command](https://github.com/gardener/etcd-backup-restore/blob/master/cmd/compact.go) to create a new snapshot.
+
+**Implementation Details:**
+
+- **Etcd Druid:**
+  - **Compaction Controller:**
+    - Introduce a new flag:
+      - **Flag:** `--hibernation-snapshot-interval`
+        - **Type:** Duration
+        - **Default:** `24h`
+        - **Description:** Interval after which a new snapshot is taken during hibernation.
+    - The compaction job starts an embedded ETCD instance to take snapshots during hibernation.
+    - This approach ensures that backups remain within the immutability period and are safeguarded against becoming mutable.
+
+###### Advantages
+
+- **No Change to ETCD Cluster State:** Does not alter the actual ETCD cluster, keeping it in hibernation.
+- **Automated Snapshot Creation:** Periodically creates new snapshots to extend immutability by triggering the compaction job.
+- **Leverages Existing Mechanism:** Utilizes the compaction job, which is already part of the system.
+
+###### Disadvantages
+
+- **Resource Consumption:** Starting an embedded ETCD instance periodically consumes resources.
+
+##### Approach 2: Re-upload of the latest snapshot
+
+**Proposed Solution:**
+
+A new `EtcdOperatorTask` called `EtcdSnapshotImmutabilityExtension` will be created as defined in the operator tasks framework. This new `EtcdOperatorTask` extends the immutability period by deploying a job which uploads another copy of the latest snapshot to the object store.
+
+A full snapshot is taken before hibernating the ETCD cluster. This is to ensure that no state maintained in the etcd cluster is lost before hibernation.
```

Comment:
Since a full snapshot is triggered before hibernation, then compaction job approach is redundant. We have decided to remove it as mentioned https://github.com/gardener/etcd-druid/pull/884#discussion_r1871190971